### PR TITLE
Darwin support

### PR DESCRIPTION
**Pull request by dedok**

### CODE REVIEW

File: build/config [hunk 1]
```diff
@@ -7,10 +7,28 @@ SOPHIA_VMINOR   = 2
 SOPHIA_BUILD   := $(shell git rev-parse --short HEAD)
 SOPHIA_DIRS     = rt version mvcc log database index repository sophia
 SOPHIA_STATIC   = libsophia.a
-SOPHIA_DSOLIB   = libsophia.so.$(SOPHIA_VMAJOR).$(SOPHIA_VMINOR)
-SOPHIA_DSO      = libsophia.so
 SOPHIA_OUTPUT   = sophia
 
+ifeq ($(shell uname -s), Darwin)
```
Comment (stephenmathieson) [inline build/config]: why not use `if/else if`?

```makefile
ifeq ($(OS), Linux)
	# ...
else ifeq ($(OS), Darwin)
	# ...
endif
```

**Comment by stephenmathieson** [comment]
+1

any thoughts @pmwkaa?

**Comment by pmwkaa** [comment]
Sure, but i think this approach is outdated for now. I've moved sophia for amalgamation build by default (generate single .c file from codebase, like sqlite does). This should highly simplify adding new build scripts like cmake, autotools or not add them at all (rely project which uses sophia). I hope this should stop the war regarding which build system is better :)

**Comment by jwerle** [comment]
@pmwkaa I'm getting *really* excited !! The dev branch is looking good

**Comment by pmwkaa** [comment]
Thanks a lot, i'm really excited too) It is coming.

**Comment by stephenmathieson** [comment]
an amalgamation build would be great!

**Comment by pmwkaa** [comment]
Thank you :)

**Pull request closed without merging**